src/backend/access/transam/xlog.c

```diff
@@ -8014,7 +8009,10 @@ static void
 PreCheckPointGuts(int flags)
 {
 	if (flags & CHECKPOINT_IS_SHUTDOWN)
+	{
 		CheckPointReplicationState(flags);
+		pgstat_write_statsfile();
```

Comment:
Please add a comment explaining why we call this here. And why it's OK that it's now called twice in shutdown sequence.

Reply:
Done